wp-content/plugins/zippicks-geo-service/includes/class-location-detector.php

```
@@ -0,0 +1,422 @@
+            if (!empty($_SERVER[$key])) {
+                $ips = explode(',', $_SERVER[$key]);
+                $ip = trim($ips[0]);
+                
+                if (filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_NO_PRIV_RANGE)) {
+                    return $ip;
+                }
+            }
+        }
+        
+        return $_SERVER['REMOTE_ADDR'] ?? '';
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'get_client_ip'. (Unused Code Rules)

(UnusedPrivateMethod)